### ISSUE TITLE
Many unit tests do not pass

### ISSUE THREAD

**Issue by Green7** [zeromq/netmq#662]
### Environment
    NetMQ Version:    
     last sources from github (20.0.4.2017)
    Operating System: 
     Windows 10, Visual Studio 2017
    .NET Version:
    4.5.1

### Actual behaviour

Many unit tests fail. 
Most of with "Cannot dispose poller from the poller thread." exception.
Some tests will never end. For example all test from class ProxyTests or InprocRouterDealerTest from 
SocketTests. It seems that they are hanging on NetMQConfig.Cleanup(); 

Is this normal behavior with current source code?

![capture](https://cloud.githubusercontent.com/assets/11303447/25228960/b0e2bfa2-25ce-11e7-93df-befc961547fd.PNG)

**Comment by drewnoakes** [comment]
Some of those errors were fixed in #666.

Seems the CI systems (AppVeyor and TravisCI) aren't finding our tests since upgrading to VS2017's csproj format, so we're not seeing these bugs on PRs. I'm working on that in #665.

```text
dotnet test src\NetMQ.Tests\NetMQ.Tests.csproj --configuration Release --no-build
Test run for C:\projects\netmq-2bhss\src\NetMQ.Tests\bin\Release\net451\NetMQ.Tests.dll(.NETFramework,Version=v4.5.1)
Microsoft (R) Test Execution Command Line Tool Version 15.0.0.0
Copyright (c) Microsoft Corporation.  All rights reserved.
The path 'C:\projects\netmq-2bhss\src\NetMQ.Tests\bin\Release\net451' specified in the 'TestAdapterPath' does not contain any test adapters, provide a valid path and try again.
Starting test execution, please wait...
No test discoverer is registered to perform discovery of test cases. Register a test discoverer and try again.
Test run for C:\projects\netmq-2bhss\src\NetMQ.Tests\bin\Release\netcoreapp1.0\NetMQ.Tests.dll(.NETCoreApp,Version=v1.0)
Microsoft (R) Test Execution Command Line Tool Version 15.0.0.0
Copyright (c) Microsoft Corporation.  All rights reserved.
Starting test execution, please wait...
No test discoverer is registered to perform discovery of test cases. Register a test discoverer and try again.
```

We are using NUnit, which seems to be a bit behind on supporting newer .NET tooling and platforms. I'm investigating moving to xUnit which can handle these things and seems to be the more popular choice nowadays.

**Comment by drewnoakes** [comment]
This should be fixed now. Tests are running again on AppVeyor, so we shouldn't have regressions like this again.
